starknet_py/net/full_node_client.py

```diff
@@ -474,11 +473,32 @@ def _create_broadcasted_txn_common_properties(transaction: AccountTransaction) -
         "type": "INVOKE"
         if transaction.tx_type == TransactionType.INVOKE_FUNCTION
         else transaction.tx_type.name,
-        "max_fee": hash_to_felt(transaction.max_fee),
-        "version": hash_to_felt(transaction.version),
-        "signature": [hash_to_felt(sig) for sig in transaction.signature],
-        "nonce": hash_to_felt(transaction.nonce)
+        "max_fee": _to_rpc_felt(transaction.max_fee),
+        "version": _to_rpc_felt(transaction.version),
+        "signature": [_to_rpc_felt(sig) for sig in transaction.signature],
+        "nonce": _to_rpc_felt(transaction.nonce)
         if transaction.nonce is not None
-        else None,
+        else "0x00",
     }
     return broadcasted_txn_common_properties
+
+
+def _to_storage_key(key: int) -> str:
+    hashed_key = hex(key).lstrip("0x")
+
+    if hashed_key[0] not in ("0", "1", "2", "3", "4", "5", "6", "7"):
+        hashed_key = "0" + hashed_key
+
+    if not 0 < len(hashed_key) <= 63:
+        raise ValueError(f"Value {key} cannot be represented as RPC storage key.")
```

Comment:
Let's say why it can't be represented (add a comment about the length)

Reply:
I'll be changing the check to verify agains the RPC regex as per @drknzz suggestion, so that kind of error won't be possible.